action/protocol/poll/governance_protocol.go

```diff
@@ -133,17 +142,25 @@ func (p *governanceChainCommitteeProtocol) CreatePreStates(ctx context.Context,
 	blkCtx := protocol.MustGetBlockCtx(ctx)
 	rp := rolldpos.MustGetProtocol(bcCtx.Registry)
 	epochNum := rp.GetEpochNum(blkCtx.BlockHeight)
+	epochStartHeight := rp.GetEpochHeight(epochNum)
 	epochLastHeight := rp.GetEpochLastBlockHeight(epochNum)
 	nextEpochStartHeight := rp.GetEpochHeight(epochNum + 1)
 	hu := config.NewHeightUpgrade(&bcCtx.Genesis)
 	if blkCtx.BlockHeight == epochLastHeight && hu.IsPost(config.Easter, nextEpochStartHeight) {
 		// if the block height is the end of epoch and next epoch is after the Easter height, calculate blacklist for kick-out and write into state DB
-		unqualifiedList, err := p.calculateKickoutBlackList(ctx, epochNum+1)
+		unqualifiedList, err := p.calculateKickoutBlackList(ctx, sm, epochNum+1)
 		if err != nil {
 			return err
 		}
-
-		return setKickoutBlackList(sm, unqualifiedList, epochNum+1)
+		return setKickoutBlackList(sm, unqualifiedList)
+	} else if blkCtx.BlockHeight == epochStartHeight && hu.IsPost(config.Easter, epochStartHeight) {
+		if err := shiftCandidates(sm); err != nil {
+			return err
+		}
+		if blkCtx.BlockHeight == 1 {
+			return nil
+		}
+		return shiftKickoutList(sm)
```

Comment:
make sure the return height of shiftKickoutList and shiftCandidates are identical